### PR TITLE
[Merged by Bors] - feat(Algebra/Module): presentation of the restriction of scalars of a module

### PR DESCRIPTION
**Pull request by joelriou**
Given a morphism of rings `A → B` and a `B`-module `M`, we obtain a presentation of `M` as a `A`-module from a presentation of `M` as `B`-module, a presentation of `B` as a `A`-module (and some additional data).

---
- [x] depends on: #18408
- [x] depends on: #18374
- [x] depends on: #18359
- [x] depends on: #18332
- [x] depends on: #18327

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Module/Presentation/RestrictScalars.lean [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    presM.exact presM.surjective_π (by
+      ext v
+      dsimp
+      simp only [Submodule.mem_top, iff_true]
+      apply Finsupp.induction
+      · simp
+      · intro r b w _ _ hw
+        refine Submodule.add_mem _ ?_ hw
+        obtain ⟨β, rfl⟩ := presB.surjective_π b
+        apply Finsupp.induction (p := fun β ↦ (Finsupp.single r (presB.π β) ∈ _))
```
Comment (joelriou) [inline Mathlib/Algebra/Module/Presentation/RestrictScalars.lean:56]:
```suggestion
        apply Finsupp.induction (p := fun β ↦ Finsupp.single r (presB.π β) ∈ _)
```

File: Mathlib/Algebra/Module/Presentation/RestrictScalars.lean [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+of `M` as a `A`-module from a presentation of `M` as `B`-module,
+a presentation of `B` as a `A`-module (and some additional data).
+
+## TODO
+* deduce that if `B` is a finitely presented as an `A`-module and `M` is
+finitely presented as an `B`-module, then `M` is finitely presented as an `A`-module
+
+-/
+
+universe v u₁ u₂
```
Comment (riccardobrasca) [inline Mathlib/Algebra/Module/Presentation/RestrictScalars.lean:22]: Can you use `Type*` instead of naming the universes?

**Review by riccardobrasca — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/Algebra/Module/Presentation/RestrictScalars.lean (hunk 2 above)
Comment (joelriou) [reply Mathlib/Algebra/Module/Presentation/RestrictScalars.lean]: Yes.

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**